### PR TITLE
Update Elevate Recurring Gifts When Info Is Updated in Batch Gift Entry

### PR DESCRIPTION
**Pull request by daniel-fuller**
This work allows a user to switch between a recurring and one-time Elevate-connected gift in batch gift entry before processing the gift in the batch.

[W-11871096](https://gus.lightning.force.com/a07EE000018D91fYAC)

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by daniel-fuller** [comment]
We are still waiting on a label approval from @ErinWiedemer in this one

**Review by npsp-reedestockton — APPROVED** [review]
Looks good, @daniel-fuller. I appreciate the fact that you always improve what you touch!

**Comment by ErinWiedemer** [comment]
> 

@daniel-fuller Stephen Cox is reviewing the error message now and I hope that we'll have an approval tomorrow. Right now, he's leaning toward the following text for the error message: You can’t edit this Elevate-connected gift until you re-enter the payment information. Select Edit Payment Information. I'll let you know as soon as I get confirmation from CCX.

**Review by ErinWiedemer — COMMENTED** [review]
I don't have the final text for the error message from CCX UI Text Review yet, but I hope to have it tomorrow. I'll Slack you as soon as I get it. But the message will be different from what is in this PR.

File: force-app/main/default/lwc/geFormRenderer/geFormRenderer.js
Comment (voduyemi) [inline force-app/main/default/lwc/geFormRenderer/geFormRenderer.js:1001]: Should we be calling this function in the `if` clause above? is there a reason for us to null these payment fields if the gift should not be removed from the Elevate Batch?

**Review by voduyemi — COMMENTED** [review]
Looks good @daniel-fuller , I just have a comment about when we null the fields

File: force-app/main/default/lwc/geFormRenderer/geFormRenderer.js
Comment (daniel-fuller) [reply force-app/main/default/lwc/geFormRenderer/geFormRenderer.js:1001]: @voduyemi I was thinking we would also want to null these fields if a user switches gift types for non-Elevate gifts as well

Comment (voduyemi) [reply force-app/main/default/lwc/geFormRenderer/geFormRenderer.js:1001]: That makes sense

**Review by voduyemi — APPROVED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (lparrott) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:5852]: Nitpick, but I like to avoid apostrophes in Custom Labels (especially this weird rich-text version!)

File: force-app/main/default/lwc/geElevateBatch/geElevateBatch.js
Comment (lparrott) [inline force-app/main/default/lwc/geElevateBatch/geElevateBatch.js:5]: I think we removed one unused `import`, then added another?

File: force-app/main/default/lwc/geGift/__tests__/geGift.test.js
Comment (lparrott) [inline force-app/main/default/lwc/geGift/__tests__/geGift.test.js:34]: These tests are great!

File: force-app/main/default/lwc/geGift/geGift.js
Comment (lparrott) [inline force-app/main/default/lwc/geGift/geGift.js:202]: Wouldn't this last condition allow an imported RD to be removed from the Batch, and is that ok? I only ask because we explicitly check for a Status of **not** Imported in the first condition.

Comment (daniel-fuller) [reply force-app/main/default/lwc/geGift/geGift.js]: @lparrott You're right. Technically, I don't think it would be a problem since it's already been processed at that point and the Update button is disabled for Imported but I updated the parentheses to be more accurate

File: force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (daniel-fuller) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml]: updated

**Review by lparrott — APPROVED** [review]

**Review by ErinWiedemer — APPROVED** [review]
LGTM!

**Pull request merged**
